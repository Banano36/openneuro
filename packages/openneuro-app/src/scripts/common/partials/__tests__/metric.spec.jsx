import React from 'react'
import { shallow } from 'enzyme'
import Metric from '../metric.jsx'

describe('Metric component', () => {
  it('renders with common props', () => {
    expect(
      shallow(<Metric type="stars" value={15} display snapshot />),
    ).toMatchSnapshot()
  })
  it('renders for downloads', () => {
    expect(
      shallow(<Metric type="downloads" value={15} display />),
    ).toMatchSnapshot()
  })
  it('renders for followers', () => {
    expect(
      shallow(<Metric type="followers" value={15} display />),
    ).toMatchSnapshot()
  })
  it('renders for views', () => {
    expect(
      shallow(<Metric type="views" value={15} display />),
    ).toMatchSnapshot()
  })
  it('throws an error for unknown types', () => {
    expect(() =>
      shallow(<Metric type="spaghetti" value={15} display />),
    ).toThrowError()
  })
})
