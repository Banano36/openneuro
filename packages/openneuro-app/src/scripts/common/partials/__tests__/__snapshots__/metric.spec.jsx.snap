// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Metric component renders for downloads 1`] = `
<span
  className="metric"
>
  <span>
    <span
      className="dataset-status ds-primary"
    >
      <TooltipTop
        tooltip="This dataset has 15 downloads."
      >
        <span
          className="icon-wrap"
        >
          <i
            className="fa fa-download"
          />
          15
        </span>
      </TooltipTop>
    </span>
  </span>
</span>
`;

exports[`Metric component renders for followers 1`] = `
<span
  className="metric"
>
  <span>
    <span
      className="dataset-status ds-primary"
    >
      <TooltipTop
        tooltip="This dataset has 15 followers."
      >
        <span
          className="icon-wrap"
        >
          <i
            className="fa fa-users"
          />
          15
        </span>
      </TooltipTop>
    </span>
  </span>
</span>
`;

exports[`Metric component renders for views 1`] = `
<span
  className="metric"
>
  <span>
    <span
      className="dataset-status ds-primary"
    >
      <TooltipTop
        tooltip="This dataset has 15 views."
      >
        <span
          className="icon-wrap"
        >
          <i
            className="fa fa-eye"
          />
          15
        </span>
      </TooltipTop>
    </span>
  </span>
</span>
`;

exports[`Metric component renders with common props 1`] = `
<span
  className="metric"
>
  <span>
    <span
      className="dataset-status ds-primary"
    >
      <TooltipTop
        tooltip="This snapshot has 15 likes."
      >
        <span
          className="icon-wrap"
        >
          <i
            className="fa fa-star"
          />
          15
        </span>
      </TooltipTop>
    </span>
  </span>
</span>
`;
